src/tiled/document.cpp

```diff
@@ -151,15 +151,17 @@ void Document::addComponent(const QList<Object *> &objects, const QString &name,
             object->addComponent(name, properties);
     }
 
-    emit componentAdded(objects, name);
+    if (objects.size() > 0)
```

Comment:
```suggestion
    if (!objects.isEmpty())
```
But why did you add this check? It's never expected to be an empty list, right?